### PR TITLE
Added option to clear cache by path

### PR DESCRIPTION
**Pull request by jacknm**
This PR adds an option to clear cache by path. It also updates the clear cache page description text to make it easier to understand the differences between the available options.

Default View
default

After Cache Cleared
after clear cache

### CODE REVIEW

File: inc/tpl/firewall-clearcache.html.tpl [hunk 1]
```diff
@@ -20,6 +20,30 @@ jQuery(document).ready(function ($) {
...
+        $.post('%%SUCURI.AjaxURL.Firewall%%', {
+            action: 'sucuriscan_ajax',
+            sucuriscan_page_nonce: '%%SUCURI.PageNonce%%',
+            form_action: 'firewall_clear_cache',
+            path
+        }, function (data) {
+            $('#firewall-clear-cache-response').html(data);
+            button.html('{{Clear Cache}}');
+        });
+    });
```
Comment (bfeddern-godaddy) [inline inc/tpl/firewall-clearcache.html.tpl:46]: You've got me reviewing... _double-checks_... jQuery?  WHAT YEAR IS IT?!?!  😜

File: inc/tpl/firewall-clearcache.html.tpl [hunk 2]
```diff
@@ -20,6 +20,30 @@ jQuery(document).ready(function ($) {
             $('#firewall-clear-cache-response').html(data);
         });
     });
+    
+    $('#firewall-clear-cache-path-button').on('click', function (event) {
+        event.preventDefault();
+
+        var button = $(this),
```
Comment (bfeddern-godaddy) [inline inc/tpl/firewall-clearcache.html.tpl:27]: is that... a `var` usage?  see comment where i ask what year this is.  i think i woke up in the early 2000s.

File: inc/tpl/firewall-clearcache.html.tpl [hunk 3]
```diff
@@ -50,14 +81,32 @@ jQuery(document).ready(function ($) {
...
+            </form>
+        </div>
+        <div id="firewall-clear-cache-auto" class="mt-2">
+            <h3 class="lead">Clear Cache Globally</h3>
+            <p>This option allows you to purge all of your page and files cache at once.</p>
+            <p>You can select the checkbox below to clear your website cache globally every time a save is made on your WordPress website.</p>
             <label>
                 <input type="checkbox" name="sucuriscan_auto_clear_cache" value="true" %%SUCURI.FirewallAutoClearCache%% />
-                <span>{{Clear cache when a post or page is updated}}</span>
+                <span>{{Clear cache whenever a post or page is updated}}</span>
```
Comment (bfeddern-godaddy) [inline inc/tpl/firewall-clearcache.html.tpl:105]: see, i told you that you knew english better than me!

File: src/firewall.lib.php [hunk 4]
```diff
@@ -665,10 +665,14 @@ public static function deblocklistAjax()
      * @param  array|bool $api_key The firewall API key.
      * @return string|bool         Message explaining the result of the operation.
      */
-    public static function clearCache($api_key = false)
+    public static function clearCache($api_key = false, $path=null)
```
Comment (bfeddern-godaddy) [inline src/firewall.lib.php:668]: should we maybe adjust the spacing on this variable declaration to be consistent?

**Review by bfeddern-godaddy — APPROVED** [review]
Looks good to me, but might be wise to wait for a review from @gvre before merging and deploying.

File: inc/tpl/firewall-clearcache.html.tpl (hunk 3 above)
Comment (jacknm) [reply inc/tpl/firewall-clearcache.html.tpl:105]: He he all credits go to Rianna :)

**Review by gvre — APPROVED** [review]
GJ 👍

**Review by gvre — APPROVED** [review]

**Pull request merged**